src/gui/title_bar.cpp

```diff
@@ -0,0 +1,600 @@
+			config.show_cycles = true;
+			continue;
+		}
+
+		if (iequals(setting, "dosbox")) {
+			check_not_set_dosbox();
+			config.show_dosbox = true;
+			continue;
+		}
+
+		if (iequals(setting, "program") ||
+		    iequals(setting, "program=name")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Name;
+			continue;
+		}
+
+		if (iequals(setting, "program=path")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Path;
+			continue;
+		}
+
+		if (iequals(setting, "program=segment")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Segment;
+			continue;
+		}
+
+		if (iequals(setting, "version") ||
+		    iequals(setting, "version=simple")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Simple;
+			continue;
+		}
+
+		if (iequals(setting, "version=detailed")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Detailed;
+			continue;
+		}
+
+		LOG_WARNING("SDL: Unknown setting '%s' in 'titlebar_content'",
+		            setting.c_str());
+	}
+
+	config.titlebar_text = conf->Get_string("titlebar_text");
+	trim(config.titlebar_text);
+}
+
+void TITLEBAR_AddConfig(Section_prop &secprop)
+{
+	constexpr auto always = Property::Changeable::Always;
+
+	Prop_string* prop_str = nullptr;
+
+	prop_str = secprop.Add_string("titlebar_content", always,
+	                              "animation cycles dosbox program=name");
+	prop_str->Set_help(
+	        "Space separated list of information to be displayed on the window title bar\n"
+	        "('animation cycles dosbox program=name' by default). Possible information are:\n"
+	        "  animation:          Animates title bar audio/video recording mark. Results\n"
+	        "                      might depend on your screen font.\n"
+	        "  cycles:             Show CPU cycles setting.\n"
+	        "  dosbox:             DOSBox emulator name.\n"
+	        "  program[=<value>]:  Display the name of a running program instead of\n"
+	        "                      'titlebarbar_text'. The <value> can be one of:\n"
+	        "                        name:     Program name, with file extension (default).\n"
+	        "                        path:     Name, extension, and full absolute path.\n"
+	        "                        segment:  Displays program memory segment name.\n"
+	        " 	                  Note: With some software (like Windows 3.1x in enhanced\n"
+	        "                      mode) it is impossible to recognize the full program name\n"
+	        "                      or path; in such case 'segment' value is used instead.\n"
+	        "  version[=<value>]:  Display DOSBox version information after emulator name.\n"
+	        "                      The <value> can be one of:\n"
+	        "                        simple:    Simple version format (default).\n"
+	        "                        detailed:  Include GIT hash, if available.");
+
+	prop_str = secprop.Add_string("titlebar_text", always, "");
```

Comment:
Technically, this should be `titlebar_custom_program_name` or something. The current name is super confusing—"text" is like "data" or "value", it can mean literally anything 😄

Reply:
> But I wonder why this setting is needed. What's the use case, who would want this and why?

Me, to make testing less confusing:) It is helpful to see whether I'm playing the pure game, or game with mods, or which test installation of Windows is currently running - I am putting all this information into the title bar, with .BAT scripts. Therefore, I definitely don't want the `"` character as a part of the syntax, as this becomes confusing when you write a batch file.

Also, with prepackaged games, it is much nicer to display the name of the game in the title bar, and not something cryptic.

I'll think about the syntax, for now I have no better idea than a separate config option.

Reply:
Thanks for the clarification. You're quite particular about the contents of your titlebar 😄